ecl/hql/hqlgram2.cpp

```diff
@@ -8992,6 +8992,34 @@ bool HqlGram::convertAllToAttribute(attribute &atr)
     return true;
 }
 
+void HqlGram::setPluggableModeExpr(attribute & targetAttr, attribute & pluginAttr, attribute & options)
+{
+    IAtom * fileFormatAtom = lower(pluginAttr.getId());
+    StringBuffer fileFormat(fileFormatAtom->queryStr());
```

Comment:
minor: You shouldn't need to copy it to a buffer.  

const char * fileFormatStr = fileFormatAtom->queryStr();

should be sufficient.

You could also use the string from getId() and preserve the original case of the string (and use a case-insensitive comparison)